### PR TITLE
:hammer: [dataset] Handle corrupted cached file in PersistentDataset

### LINKED ISSUE #5723: Reloading of corrupted cached files for PersistentDataset
I use PersistentDataset in my train loop. I needed to abort train script and when I run it again, one of cached files was corrupted because of abortion. PersistentDataset did not finish saving this cache file.
It threw an exception `RuntimeError: Invalid magic number; corrupt file?`

I would like to add a possibility to handle corrupt files. If cached file is corrupted, PersistenDataset will continue like there was no cached file and replace it with a good one.

I see, that if I add "except" block [here](https://github.com/Project-MONAI/MONAI/blob/2b160b3db3ee2942a28ff3a5628879c3fe46d7c8/monai/data/dataset.py#L376-L381), I can handle such behaviour.

May I take this ticket and make a PR?

### PR DESCRIPTION
**Pull request by ishandutta0098**
Fixes #5723

### Description
Corrupted cached files in the PersistentDataset cause the exception:
`RuntimeError: Invalid magic number; corrupt file?`

With this PR we handle that case in the try-except block and continue the usual functionality if the cached file was absent.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by ishandutta0098** [comment]
Requesting review: @wyli @KumoLiu

File: monai/data/dataset.py [hunk 1]
```diff
@@ -387,6 +387,12 @@ def _cachecheck(self, item_transformed):
             except PermissionError as e:
                 if sys.platform != "win32":
                     raise e
+            except RuntimeError as e:
+                if "Invalid magic number; corrupt file" in str(e):
+                    print(f"Corrupt cache file detected: {hashfile}. Deleting and recomputing.")
```
Comment (KumoLiu) [inline monai/data/dataset.py:392]: Adding a warning here would be better.
```suggestion
                    warnings.warn(f"Corrupt cache file detected: {hashfile}. Deleting and recomputing.")
```

**Review by KumoLiu — APPROVED** [review]
Thanks for the PR, LGTM.

**Pull request merged**